modules/iam-eks-role/policies.tf

```diff
@@ -0,0 +1,454 @@
+data "aws_partition" "current" {}
+
+################################################################################
+# Cluster Autoscaler Policy
+################################################################################
+
+# https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/cloudprovider/aws/README.md
+data "aws_iam_policy_document" "cluster_autoscaler" {
+  count = var.create_role && var.attach_cluster_autoscaler_policy ? 1 : 0
+
+  statement {
+    actions = [
+      "autoscaling:DescribeAutoScalingGroups",
+      "autoscaling:DescribeAutoScalingInstances",
+      "autoscaling:DescribeLaunchConfigurations",
+      "autoscaling:SetDesiredCapacity",
+      "autoscaling:TerminateInstanceInAutoScalingGroup",
+      "ec2:DescribeInstanceTypes",
+    ]
+
+    resources = ["*"]
```

Comment:
This would be too broad for most people I think as it would allow termination of ANY EC2 instance in an ASG.

Personally I would copy the policy from [here](https://github.com/terraform-aws-modules/terraform-aws-eks/pull/716/files#diff-49c22f96d12f8b3391c9493b1ebcf2d0b8f9d6948fec4f0f3475338b0b1a7c6bR26-R66) as it includes the proper `condition` to limit the policy to this cluster. But then you would need to pass in the `cluster_id`...

Reply:
Could also be done in a later PR by someone who wants it

Reply:
thats a great point, skipped over that section in the docs. this too fits inline with the other policies where the docs are more open with `"*"` but here we are defaulting to open-ish while giving folks access to be more restrictive. added in https://github.com/terraform-aws-modules/terraform-aws-iam/pull/184/commits/41d38278638f6ea2df00492b69deedfd5c0a304d